packages/loot-core/migrations/1699454123000_reports.sql

```diff
@@ -0,0 +1,25 @@
+BEGIN TRANSACTION;
```

Comment:
(disclaimer: I haven't reviewed anything yet; only making this comment for now)

For reviewers (and also @carkom): pay SPECIAL attention to this migration. The user interface can be changed and improved later on. But the database structure is **very hard** to change without causing sync problems (as we've seen in the past). So making sure this migration is what we want in the short & medium term is crucial.

Reply:
Thanks @MatissJanis . I tried to include everything I needed now and a few columns for the future (eg. sort_order, as a way for possible drag/drop ordering UI). Have I implemented it okay? It's my the first actual migration file I've created for Actual.

Reply:
What's nice is that its a new table, so hopefully there is a lot less risk in the migration w.r.t. to breaking existing things, but certainly its a good idea to try and future-proof as much as possible.

Probably i'm being dumb but does it save the range? e.g. if i want a report that always shows me the last 6 months? 
Or will it save start/end of whatever the last 6 months happened to be at the time?

is it worth adding a column for custom colours? could just be stored as a json string or something. Just trying to think of things people may want in the future.

Reply:
Yea it saves start/end. Selecting a range is just forcing a change to start/end.

Colors is a good idea. I'll add it. LMK if you think of others.

Reply:
@MatissJanis, one thing I've done before is creating some sort of "misc" or "extras" column. Then set it up as an array of objects insead of only saving one element it can be added to for any future changes. This way we can say the current structure covers everything we want now and in the medium term and for any long term changes we can add those elements to the "extras" array. With this implementation we'd never need to change the SQL table structure.

Reply:
Sorry, missed this comment.

The problem with that is: querying becomes much more inefficient. SQL is pretty useful at indexing top level attributes to achieve blazing-fast query performance. But when we talk about deep-nested arrays/objects.. the performance sharply declines.

Overall I think we don't need the `extra` column now, but I wouldn't be totally against it either.

It's always possible to add more columns later on. Removing/renaming columns, however, is the tricky part that we should try and avoid.